src/UnoFlipModel.java

```diff
@@ -382,151 +242,269 @@ private boolean isWinner(Player player){
         }
     }
 
+    /**
+     * Draw the amount of cards(n) based on which player(index) will be receiving them
+     * @param n The amount of cards to be added to the hand of the player
+     * @param index The index of the player that will be receiving cards
+     */
+    public void drawNCards(int n,int index){
+        this.players.get(index).addCardToHand(n);
+    }
+
     /**
      * Go to the turn of the next player based on turn direction
      */
     public void nextTurn() {
 
-        if (turnFinished) { //If the turn is finished, allow player to press next player
-            //clockwise
-            if (turnDirection) { //0->1->2->3
-                currentTurn = (currentTurn + 1) % numPlayers;
-                nextPlayerIndex = (currentTurn + 1) % numPlayers;
-            }
-            //counterclockwise
-            else { //0->3->2->1
-                currentTurn = (currentTurn - 1 + numPlayers) % numPlayers;
-                nextPlayerIndex = (currentTurn - 1 + numPlayers) % numPlayers;
+        //if the turn is finished, allow player to press next player
+        if (this.turnFinished) {
+
+            //clockwise (ex. 0->1->2->3)
+            if (this.turnDirection) {
+                this.currentTurn = (this.currentTurn + 1) % this.numPlayers;
+                this.nextPlayerIndex = (this.currentTurn + 1) % this.numPlayers;
+
+            //counterclockwise (ex. 0->3->2->1)
+            } else {
+                this.currentTurn = (this.currentTurn - 1 + this.numPlayers) % this.numPlayers;
+                this.nextPlayerIndex = (this.currentTurn - 1 + this.numPlayers) % this.numPlayers;
             }
-            if (skipTurn){
-                if (turnDirection) { //0->1->2->3
-                    currentTurn = (currentTurn + 1) % numPlayers;
-                    nextPlayerIndex = (currentTurn + 1) % numPlayers;
-                }
-                //counterclockwise
-                else { //0->3->2->1
-                    currentTurn = (currentTurn - 1 + numPlayers) % numPlayers;
-                    nextPlayerIndex = (currentTurn - 1 + numPlayers) % numPlayers;
+
+            //if next player's turn is being skipped (ex. player 2 is being skipped:  0->1->3->0)
```

Comment:
redundant comment. just make a comment inside the if block saying "go to the next turn again to skip a player"